src/Raven.Server/Documents/ResourceCache.cs

```diff
@@ -36,24 +44,29 @@ public sealed class ResourceDetails
         /// <summary>
         /// This locks the entire cache. Use carefully.
         /// </summary>
-        public IEnumerable<Task<TResource>> Values => _caseInsensitive.Values;
+        public IEnumerable<Task<TResource>> Values => _readonlyCaseInsensitive.Values;
 
-        public int Count => _caseInsensitive.Count;
+        public int Count => _readonlyCaseInsensitive.Count;
 
-        internal int DetailsCount => _resourceDetails.Count;
+        internal int DetailsCount => _readonlyResourceDetails.Count;
 
         public void Clear()
```

Comment:
This should not be under a lock?

Reply:
I know that it is almost never called, but still

Reply:
@redknightlois ?

Reply:
BTW:  Is `TryGetValue`  thread unsafe now? In case when you perform clean/remove in other thread,

Reply:
Those are readonly, worst case scenario you are asking to the wrong dictionary.